analytics_data_api/v0/views/learners.py

```diff
@@ -0,0 +1,88 @@
+"""
+API methods for module level data.
+"""
+from django.http import Http404
+from rest_framework.exceptions import ParseError
+from rest_framework import generics
+
+from opaque_keys import InvalidKeyError
+from opaque_keys.edx.keys import CourseKey
+
+from analytics_data_api.v0.models import RosterEntry
+from analytics_data_api.v0.serializers import LearnerSerializer
+
+
+class CourseViewMixin(object):
+    """
+    Captures the course_id query arg and validates it.
+    """
+
+    course_id = None
+
+    def get(self, request, *args, **kwargs):
+        self.course_id = request.QUERY_PARAMS.get('course_id', None)
+        if not self.course_id:
+            raise ParseError('Course ID was not specified')
+        try:
+            CourseKey.from_string(self.course_id)
+        except InvalidKeyError:
+            raise ParseError('Course ID malformed')
```

Comment:
Yes, this returns a 400, but not in the edx recommended format.  I'll see about updating that.